### PR TITLE
Support more runtime types from bluemix host info.

### PR DESCRIPTION
**Pull request by lzbj**
Or from local pre values.

### CODE REVIEW

File: tests/src/integration/common/wskdeploy.go [hunk 1]
```diff
@@ -47,17 +47,17 @@ func (wskdeploy *Wskdeploy) RunCommand(s ...string) ([]byte, error) {
 }
 
 func (wskdeploy *Wskdeploy) Deploy(manifestPath string, deploymentPath string) ([]byte, error) {
-	return wskdeploy.RunCommand("-m", manifestPath, "-d", deploymentPath)
+	return wskdeploy.RunCommand("-i ", "false", "-m", manifestPath, "-d", deploymentPath)
```
Comment (pritidesai) [inline tests/src/integration/common/wskdeploy.go]: @lzbj I am getting this error when I run wskdeploy with -i false, are you sure these tests are successful?

```
./wskdeploy -i false -m tests/src/integration/flagstests/manifest.yaml 
Error: unknown command "false" for "wskdeploy"
Run 'wskdeploy --help' for usage.
2017/08/21 16:49:19 unknown command "false" for "wskdeploy"
```

Comment (lzbj) [inline tests/src/integration/common/wskdeploy.go]: @pritidesai ,thanks for your review, yes, I think we need to change the command to
wskdeploy -i -m
Anyway, Vincent's latest PR will fix this issue I think.

Comment (mrutkows) [inline tests/src/integration/common/wskdeploy.go]: Just merged https://github.com/apache/incubator-openwhisk-wskdeploy/pull/375, we now have the flag "UserInteractive" (default to false), but need to verify on this issue if it had the desired effect.

Comment (pritidesai) [inline tests/src/integration/common/wskdeploy.go]: @lzbj I think the intent here was to disable interactive mode, but now like Matt is saying it is disabled by default, so you don't need this flag "-i" here at all. If you specify "-i" now on wskdeploy CMD line, interactive mode is turned on.

Comment (lzbj) [inline tests/src/integration/common/wskdeploy.go]: ok, got it, correct, I'll update the code.

File: tests/src/integration/runtimetests/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package:
```
Comment (mrutkows) [inline tests/src/integration/runtimetests/manifest.yml]: @lzbj I do not see tests fro Python2 and Python3, which are both supported currently by OpenWhisk.  i.e., 
"image": "openwhisk/python2action:latest",
"deprecated": false,
"requireMain": false,
"default": true,
"attached": false,
"kind": "python:2"
}, {
"image": "openwhisk/python3action:latest",
"deprecated": false,
"requireMain": false,
"default": false,
"attached": false,
"kind": "python:3"
}],

File: tests/src/integration/runtimetests/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            inputs:
+              name: string
+              place: string
+            outputs:
+              payload: string
+  actions:
+          helloworldjava:
+            function: src/hello.jar
+            runtime: java
+            main: Hello
```
Comment (mrutkows) [inline tests/src/integration/runtimetests/manifest.yml]: @lzbj Should we test an invalid value to test our response? 

Also, I would like to see a valid "kind" (like NodeJS) used with an old (no longer supported) version to test for that as well.

**Comment by mrutkows** [comment]
Intended to fix issue https://github.com/apache/incubator-openwhisk-wskdeploy/issues/306

File: tests/src/integration/runtimetests/manifest.yml (hunk 2 above)
Comment (mrutkows) [reply tests/src/integration/runtimetests/manifest.yml]: We should likely test for python:2, python:3, nodejs:5 and nodejs:6 (and what happens if we use python2.7 (point versions)?

File: tests/src/integration/runtimetests/manifest.yml (hunk 3 above)
Comment (lzbj) [reply tests/src/integration/runtimetests/manifest.yml]: ok, I'll add more test.

File: tests/src/integration/runtimetests/manifest.yml (hunk 2 above)
Comment (lzbj) [reply tests/src/integration/runtimetests/manifest.yml]: ok, add test

File: tests/src/integration/jaraction/manifest.yaml [hunk 4]
```diff
@@ -1,7 +0,0 @@
-package:
```
Comment (mrutkows) [inline tests/src/integration/jaraction/manifest.yaml]: @lzbj Why did we delete this manifest?

**Comment by mrutkows** [comment]
@lzbj The Travis has a failure (and reports PASS) for the new tests:
```
=== RUN   TestSupportProjectPath

wskdeploy.Path is wskdeploy

==> Executing: wskdeploy -m /home/travis/gopath/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/runtimetests

==> Output: 2017/08/29 07:53:00 Unmarshal Openwhisk info from internet

arg is -m

{"error":"Missing cmd key"}

--- PASS: TestSupportProjectPath (0.59s)

PASS

ok  	github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/runtimetests	0.602s
```

**Comment by pritidesai** [comment]
@lzbj `TestSupportProjectPath` is failing with the following error, its executing with `-m` option but argument points to a directory.

```
go test ./tests/src/integration/runtimetests/ -tags=integration -run=TestSupportProjectPath -v
=== RUN   TestSupportProjectPath
wskdeploy.Path is wskdeploy
==> Executing: wskdeploy -m /Users/pritidesai/Documents/goworkspace/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/runtimetests
==> Output: 2017/08/29 10:08:02 Unmarshal Openwhisk info from internet
2017/08/29 10:08:02 read /Users/pritidesai/Documents/goworkspace/src/github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/runtimetests: is a directory
2017/08/29 10:08:02 Error happened during execution, please type 'wskdeploy -h' for help messages.

==> Error: exit status 1
--- FAIL: TestSupportProjectPath (0.38s)
        Error Trace:    runtimes_test.go:36
        Error:          Not equal: 
                        expected: <nil>(<nil>)
                        actual: *exec.ExitError(&exec.ExitError{ProcessState:(*os.ProcessState)(0xc4200f5940), Stderr:[]uint8(nil)})
        Messages:       Failed to deploy based on the manifest path
FAIL
exit status 1
FAIL    github.com/apache/incubator-openwhisk-wskdeploy/tests/src/integration/runtimetests      0.396s
```

**Comment by mrutkows** [comment]
@lzbj the "jaraction" test should not have been brought into the runtimes tests. It should remain separate as it tests specifically the ability to jar an action. Please put it back and add another test for Java if needed.

**Comment by mrutkows** [comment]
@lzbj @pritidesai @houshengbo Priti has volunteered to "fix" this if I merge it today and work to put back the "jaraction" test as well so we can move forward and not lose another day.

**Comment by lzbj** [comment]
sorry, my bad, I thought i was belong to runtimes, so @pritidesai , please help fix this and I'll merged your PR asap.

**Pull request merged**